src/App.js

```diff
@@ -27,30 +37,89 @@ const App = ({ nearConfig, data }) => {
+        // We pass the public key into the `new_pk` parameter
+
+        transaction = await crosswordAccount.functionCall(
+          {
+            contractId: nearConfig.contractName,
+            methodName: 'claim_reward_new_account',
+            args: {
+              crossword_pk: solvedPuzzle,
+              new_acc_id: winner_account_id,
+              new_pk: playerKeyPair.publicKey,
+              memo
+            },
+            gas: '300000000000000', // You may omit this for default gas
+            attachedDeposit: 0  // You may also omit this for no deposit
+          }
+        );
+      } else {
+        transaction = await crosswordAccount.functionCall(
+          {
+            contractId: nearConfig.contractName,
+            methodName: 'claim_reward',
+            args: {
+              crossword_pk: solvedPuzzle,
+              receiver_acc_id: winner_account_id,
+              memo
+            },
+            gas: '300000000000000', // You may omit this for default gas
+            attachedDeposit: 0  // You may also omit this for no deposit
+          }
+        );
+      }
     } catch (e) {
+      console.error('Unexpected error when claiming', e);
       if (e.message.contains('Can not sign transactions for account')) {
         // Someone has submitted the solution before the player!
         console.log("Oof, that's rough, someone already solved this.")
       }
     } finally {
       setShowLoader(false);
+      // See if the transaction succeeded during transfer
+      // or succeeded when creating a new account.
+      // If unsuccessful, let the user try again.
       console.log('Transaction status:', transaction.status);
+      const tx_succeeded = transaction.status.hasOwnProperty('SuccessValue');
+      if (tx_succeeded) {
+        let tx_success_value = b64ToUtf8(transaction.status.SuccessValue);
+        if (needsNewAccount) {
+          // Look for base64-encoded "false"
+          if (tx_success_value === 'true') {
+            // This tells the React app that it's solved and claimed
+            setSolvedPuzzle(false);
+            setClaimError('');
+
+            // Clean up and get ready for next puzzle
+            localStorage.removeItem('playerSolvedPuzzle');
+            localStorage.removeItem('guesses');
+          } else {
+            setClaimError('Could not create that account, please try another account name.');
+          }
+        } else {
+          if (tx_success_value === 'true') {
+            // This tells the React app that it's solved and claimed
+            setSolvedPuzzle(false);
+            setClaimError('');
+            // Clean up and get ready for next puzzle
+            localStorage.removeItem('playerSolvedPuzzle');
+            localStorage.removeItem('guesses');
+          } else {
+            setClaimError("Couldn't transfer reward to that account, please try another account name or create a new one.")
+          }
+        }
```

Comment:
Could reduce the outer `if` block to the following, since the error message is the only thing dependent on `needsNewAccount`
```suggestion
        // Look for base64-encoded "false"
        if (tx_success_value === 'true') {
          // This tells the React app that it's solved and claimed
          setSolvedPuzzle(false);
          setClaimError('');

          // Clean up and get ready for next puzzle
          localStorage.removeItem('playerSolvedPuzzle');
          localStorage.removeItem('guesses');
        } else {
          if (needsNewAccount) {
            setClaimError('Could not create that account, please try another account name.');
          } else { 
            setClaimError("Couldn't transfer reward to that account, please try another account name or create a new one.");
          }
        }
```